lib/proposer_calculator_test.go

```diff
@@ -3,23 +3,51 @@ package sebak
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-type SelfProposerCalculator struct {
-}
-
-func (c SelfProposerCalculator) Calculate(nr *NodeRunner, _ uint64, _ uint64) string {
-	return nr.localNode.Address()
-}
-
+/*
+In TestProposerCalculator test, the proposer is always the node itself because of SelfProposerCalculator.
+*/
 func TestProposerCalculator(t *testing.T) {
 	nodeRunners := createTestNodeRunner(1)
 
 	nodeRunner := nodeRunners[0]
 	nodeRunner.SetProposerCalculator(SelfProposerCalculator{})
 
-	assert.Equal(t, nodeRunner.localNode.Address(), nodeRunner.CalculateProposer(0, 1))
-	assert.Equal(t, nodeRunner.localNode.Address(), nodeRunner.CalculateProposer(0, 2))
-	assert.Equal(t, nodeRunner.localNode.Address(), nodeRunner.CalculateProposer(1, 2))
+	require.Equal(t, nodeRunner.localNode.Address(), nodeRunner.CalculateProposer(1, 0))
+	require.Equal(t, nodeRunner.localNode.Address(), nodeRunner.CalculateProposer(2, 0))
+	require.Equal(t, nodeRunner.localNode.Address(), nodeRunner.CalculateProposer(2, 1))
+}
+
+/*
+	TestNodesHaveSameProposers tests all 3 nodes have the same proposers at each round.
```

Comment:
Side note: those names are quite long, and since the comment is right on the symbol, only add noise.
E.g.
```
// All 3 nodes have the same proposer at each round
```

is enough